src/openzeppelin/tests/test_accesscontrol.cairo

```diff
@@ -0,0 +1,125 @@
+use openzeppelin::access::accesscontrol::AccessControl;
+use openzeppelin::utils::constants::{DEFAULT_ADMIN_ROLE, IACCESSCONTROL_ID};
+use starknet::{contract_address_const, ContractAddress, testing::set_caller_address};
+
+const SOME_OTHER_ROLE: felt252 = 42;
+
+fn ACCOUNT1() -> ContractAddress {
+    contract_address_const::<1>()
+}
+
+fn ACCOUNT2() -> ContractAddress {
+    contract_address_const::<2>()
+}
+
+fn setup() {
+    set_caller_address(ACCOUNT1());
+    AccessControl::constructor(ACCOUNT1());
+}
+
+#[test]
+#[available_gas(2000000)]
+fn test_initializer() {
+    AccessControl::initializer();
+    assert(AccessControl::supports_interface(IACCESSCONTROL_ID), 'Should support own interface');
+}
+
+#[test]
+#[available_gas(2000000)]
+fn test_constructor() {
+    AccessControl::constructor(ACCOUNT1());
+    assert(AccessControl::has_role(DEFAULT_ADMIN_ROLE, ACCOUNT1()), 'Admin role should be set');
+    assert(AccessControl::supports_interface(IACCESSCONTROL_ID), 'Should support own interface');
+}
+
+#[test]
+#[available_gas(2000000)]
+fn test_assert_only_role() {
+    setup();
+    AccessControl::assert_only_role(DEFAULT_ADMIN_ROLE);
+}
+
+#[test]
+#[available_gas(2000000)]
+#[should_panic(expected: ('Caller is missing role', ))]
+fn test_assert_only_role_unauthorized() {
+    setup();
+    set_caller_address(ACCOUNT2());
+    AccessControl::assert_only_role(DEFAULT_ADMIN_ROLE);
+}
+
+#[test]
+#[available_gas(2000000)]
+fn test_grant_role() {
+    setup();
+    AccessControl::grant_role(DEFAULT_ADMIN_ROLE, ACCOUNT2());
+    assert(AccessControl::has_role(DEFAULT_ADMIN_ROLE, ACCOUNT2()), 'Role should be granted');
```

Comment:
do we need to also test `has_role`? when it should return `true` and when `false`